dss-collection-nft/contracts/DSSCollection.cdc

```diff
@@ -0,0 +1,526 @@
+
+    // NFT Events
+    //
+    pub event DSSCollectionNFTMinted(
+        id: UInt64,
+        collectionGroupID: UInt64,
+        serialNumber: UInt64,
+        completedBy: String,
+        completionDate: UFix64,
+        level: UInt64
+    )
+    pub event DSSCollectionNFTBurned(id: UInt64)
+
+
+    // Named Paths
+    //
+    pub let CollectionStoragePath:  StoragePath
+    pub let CollectionPublicPath:   PublicPath
+    pub let AdminStoragePath:       StoragePath
+    pub let MinterPrivatePath:      PrivatePath
+
+    // Entity Counts
+    //
+    pub var totalSupply:                 UInt64
+    pub var nextCollectionGroupID:       UInt64
+
+    // List of collection groups
+    //
+    access(self) let collectionGroupByID: @{UInt64: CollectionGroup}
+
+    // A public struct to access CollectionGroup data
+    //
+    pub struct CollectionGroupData {
+        pub let id: UInt64
+        pub let name: String
+        pub let productPublicPath: PublicPath
+        pub let open: Bool
+        pub let startTime: UFix64?
+        pub let endTime: UFix64?
+        pub let timeBound: Bool
+        pub var editionIDInCollectionGroup: {UInt64: Bool}
+
+        pub fun editionIDExistsInCollectionGroup(editionID: UInt64): Bool {
+           return self.editionIDInCollectionGroup.containsKey(editionID)
+        }
+
+        init (id: UInt64) {
+            if let collectionGroup = &DSSCollection.collectionGroupByID[id] as &DSSCollection.CollectionGroup? {
+                self.id = collectionGroup.id
+                self.name = collectionGroup.name
+                self.productPublicPath = collectionGroup.productPublicPath
+                self.open = collectionGroup.open
+                self.startTime = collectionGroup.startTime
+                self.endTime = collectionGroup.endTime
+                self.timeBound = collectionGroup.timeBound
+                self.editionIDInCollectionGroup = collectionGroup.editionIDInCollectionGroup
+            } else {
+                panic("collectionGroup does not exist")
+            }
+        }
+    }
+
+    // A top-level CollectionGroup with a unique ID and name
+    //
+    pub resource CollectionGroup {
+        pub let id: UInt64
+        pub let name: String
+        pub let productPublicPath: PublicPath
+        pub var open: Bool
+        pub let startTime: UFix64?
+        pub let endTime: UFix64?
+        pub let timeBound: Bool
+        pub var numMinted: UInt64
+        pub var editionIDInCollectionGroup: {UInt64: Bool}
+
+        // Close this collection group
+        //
+        access(contract) fun close() {
+            pre {
+                self.open == true: "not open"
+            }
+
+            self.open = false
+
+            emit CollectionGroupClosed(id: self.id)
+        }
+
+        // Add editionID to collection group
+        //
+        access(contract) fun addEditionToCollectionGroup(editionID: UInt64) {
```

Comment:
can we make it accept an array of Edition incase we want to add multiple editions to avoid calling the function multiple times

Reply:
Yes, I moved this to slots & items, and turned it into an array.